feature_engine/model/model_builder.py

```diff
@@ -366,18 +366,31 @@ def create_holdout_sets(self, val_size: float=0.1, test_size: float=None) -> Non
         """
         # Calls the `define_dataset_for_model()` function to get the `X`s and the `y`s
         self.X, self.y = self.define_dataset_for_model()
+        
         # If we need to split out a test set
-        if test_size:
+        if self.test_dataset_names == None:
             # Spit train-val and test set in a `(1-test_size)-test_size` split
-            X_train_val, X_test, y_train_val, y_test = train_test_split(self.X, self.y, random_state=42, test_size=test_size)
+            X_train_val = self.X
+            y_train_val = self.y
+            
+            # Split test set only if specified
+            if test_size: 
```

Comment:
testing for the special case here, where we actually need the separate test dataset